File: app/src/main/java/kg/koronastaff/staffapp/ui/quarantine/QuarantineFragment.java
package kg.koronastaff.staffapp.ui.quarantine;

import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import kg.koronastaff.staffapp.R;

public class QuarantineFragment extends Fragment {

    private QuarantineViewModel mViewModel;

    public static QuarantineFragment newInstance() {
        return new QuarantineFragment();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_quarantine, container, false);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        mViewModel = new ViewModelProvider(this).get(QuarantineViewModel.class);
        // TODO: Use the ViewModel
    }

}

File: app/src/main/java/kg/koronastaff/staffapp/ui/quarantine/QuarantineFragment.kt
package kg.koronastaff.staffapp.ui.quarantine

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import kg.koronastaff.staffapp.R
import kg.koronastaff.staffapp.adapters.QuarantineAdapter
import kg.koronastaff.staffapp.models.QuarantineSteps
import kg.koronastaff.staffapp.ui.FragmentWithStat
import kotlinx.android.synthetic.main.fragment_quarantine.*
import java.util.*

class QuarantineFragment : FragmentWithStat() {
    private var recyclerView: RecyclerView? = null
    private var mAdapter: QuarantineAdapter? = null
    private var viewManager: RecyclerView.LayoutManager? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        super.onCreateView(inflater, container, savedInstanceState)
        val root = inflater.inflate(R.layout.fragment_quarantine, container, false)
        recyclerView = root.findViewById(R.id.fake_news_recycler)
        viewManager = LinearLayoutManager(context)

        mAdapter = QuarantineAdapter(arrayListOf())

        recyclerView = root.findViewById<RecyclerView>(R.id.fake_news_recycler).apply {
            setHasFixedSize(true)
            layoutManager = viewManager
            adapter = mAdapter
        }

        super.coronaViewModel.getQuarantineSteps()?.subscribe{
            if (it.results!!.size < 1){
                it.results = arrayListOf()
            }
            textView6.visibility = View.GONE
            mAdapter!!.update(it.results!!)
        }

        return root
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        super.updateStats(cache.getStat())
    }
}